### PR TITLE
chore: Add documentations for superfluid module

### PR DESCRIPTION
**Pull request by mattverse**
sub component of: #1955

## What is the purpose of the change

This PR adds proto message documentation and additional documentation for msg_server and code within `x/superfluid/keeper`. 

This PR does not include ANY change or refactoring for any of the logic within `x/superfluid`

## Brief Changelog
- Add documentation for `x/superfluid`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? additional documentations

### CODE REVIEW

File: proto/osmosis/superfluid/params.proto
Comment (mattverse) [reply proto/osmosis/superfluid/params.proto:11]: Elabortaion added!

File: proto/osmosis/superfluid/genesis.proto
Comment (mattverse) [reply proto/osmosis/superfluid/genesis.proto:13]: Fixed all changes to proto cases!

File: x/superfluid/keeper/msg_server.go
Comment (mattverse) [reply x/superfluid/keeper/msg_server.go:42]: Addressed!

**Review by p0mvn — APPROVED** [review]

File: proto/osmosis/superfluid/genesis.proto
Comment (ValarDragon) [inline proto/osmosis/superfluid/genesis.proto:18]:
```suggestion
  // osmo_equivalent_multipliers is the records of osmo equivalent amount of
  // each superfluid registered pool, updated every epoch.
```

File: proto/osmosis/superfluid/params.proto
Comment (ValarDragon) [inline proto/osmosis/superfluid/params.proto:13]:
```suggestion
  // superfluid staking, default: 5%. The minimum risk factor works 
  // to counter-balance the staked amount on chain's exposure to various asset volatilities,
  // and have base staking be 'resistant' to volatility.
```

**Review by ValarDragon — APPROVED** [review]
Nice job! Good to merge imo once typo fixes are in

**Comment by mattverse** [comment]
L forgot to push commit's with @ValarDragon 's recent comments before merging this, apologies for the confusion, creating a secondary PRs with the fixes :(

**Pull request merged**
